## App Store Analysis

The goal of this analysis will be to take two data sets, Google Play and iOS App Store and break down what types of apps user's are most attracted to. 

The Google Play data was collected in August 2018 and contains data on approximately 10,000 Android Apps.
The iOS data was collected in July 2017 and contains approximately data on 7,000 iOS Apps.

We will be focusing on apps designed for an english speaking audience.
As well as only free to download apps. 

In [3]:
from csv import reader

# Open Google Play Store Data
# add encoding to avoid error "'charmap' codec can't decode byte 0x90 in position"
# Use r before the directory location to convert to a raw string
opened_file = open(r"C:\Users\MalikSami\Desktop\Data_Analytics\App_Store_Analysis\googleplaystore.csv",
                  encoding="utf8")
read_file = reader(opened_file)

android = list(read_file)
android_header = android[0]
android = android[1:]


In [5]:
# Open iOS App Store Data
opened_file = open(r"C:\Users\MalikSami\Desktop\Data_Analytics\App_Store_Analysis\AppleStore.csv",
                  encoding="utf8")
read_file = reader(opened_file)

ios = list(read_file)
ios_header = ios[0]
ios = ios[1:]

In [6]:
# Explore Function 
# start and end are integers that represent the location of the splices 
# rows_and_columns will priint the number of rows and columns if set to True
# Note the data set should not have a header row, or the function will print the wrong number of rows
def explore_data(dataSet, start, end, rows_and_columns = False):
    dataSet_slice = dataSet[start:end]
    for row in dataSet_slice:
        print(row)
        print('\n')
    if rows_and_columns:
        print('Number of rows: ', len(dataSet))
        print('Number of columns: ', len(dataSet[0]))
        
    

### Exploring the data

In [9]:
explore_data(android, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


Number of rows:  10841
Number of columns:  13


In [10]:
explore_data(ios, 0, 3, True)

['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


Number of rows:  7197
Number of columns:  16


#### We can look at the header row of each data set to find relevant columns that can be useful in our analysis

In [13]:
print(android_header)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


##### In the Android data set we should consider looking at:
'App' , 'Category', 'Rating', 'Reviews', 'Price', 'Content Rating' and 'Genres'



In [16]:
print(ios_header)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


##### In the iOS data set we should consider looking at:
'track_name', 'price', 'rating_count_tot',  'user_rating', 'cont_rating', 'prime_genre'

rating_count_tot = total number of ratintgs
cont_rating = Recomended Age

## Data Cleaning

Since we are only analyzing apps that are in english and free, we need to clean our data. 
Also if there are any errors in the data, those need to be addressed. 


Here We can see by running a for loop on the android data, that the header length is not equal to the length of the row. The following code pin points where the error is. 
From here we can delete that row from our data. Make sure to only run the del statement once since it will delete the next row if run again. 

In [7]:
for row in android:
    header_length = len(android_header)
    row_length = len(row)
    if row_length != header_length:
        print(row)
        print(android.index(row))

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']
10472


In [8]:
print(len(android))
del(android[10472])
print(len(android))


10841
10840


#### Duplicate Data
Next we need to search for duplicate data in our data set. 
We can create two lists, duplicate_apps and unique_apps and run a for loop to seperate the data into the two lists.

In [9]:
duplicate_apps = []
unique_apps = []

for app in android:
    name = app[0]
    if name in unique_apps:
        duplicate_apps.append(name)
    else:
        unique_apps.append(name)

print('Number of duplicate apps: ', len(duplicate_apps))
print('Number of unique apps: ', len(unique_apps))

Number of duplicate apps:  1181
Number of unique apps:  9659


We do not want to randomly delete the duplicates. Instead we want to find a way to keep the most recent data and get rid of the older duplicates. If you note the 4th index of the data, which is the number of reviews. We can see that that a higher total number of reviews would mean that the data is more recent.

In [10]:
# Note the fourth index, which is total reviews.
for app in android:
    name = app[0]
    if name == 'Instagram':
        print(app)

print('Expected length: ', len(android) - 1181)

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
Expected length:  9659


Now we can create a dictionary and if the app name is already in the dictionary and the app's total reviews are greater than the n_reviews, we update the number of reviews for that entry.

If the app name is not in the dictionary than we create a new entry in the dictionary, where the key is the app name and the value is the total number of reviews.

In [11]:
reviews_max = {}

for app in android:
    name = app[0]
    n_reviews = float(app[3])
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews
        
    elif name not in reviews_max:
        reviews_max[name] = n_reviews

print(len(reviews_max))
        
    

9659


Now we can create a list of a cleaned version of the android data (android_clean)

We can write a for loop to  run and see if reviews_max[name] is == n_reviews and that it does not already exist in the already_added list.
If both conditions are met, the row is added to android_clean and the app name is added to the already_added list

In [12]:
# We store rows in android_clean
android_clean = []
# We store the app name in already_added
already_added = []

for app in android:
    name = app[0]
    n_reviews = float(app[3])
    if (reviews_max[name] == n_reviews) and (name not in already_added):
        android_clean.append(app)
        already_added.append(name)

        
    

In [24]:
explore_data(android_clean,0,3,True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows:  9659
Number of columns:  13


#### Cleaning the non - English apps

Now we can write a function to take out the non  english apps from the data. 

The function will take a string and use the built in python function ord(), which returns a integer representing the unicode character.
If that code is greater than 127 then that would mean it is not an english character. 




In [13]:
def is_english(string):
    
    for character in string:
        if ord(character) > 127:
            return False
        
    return True

print(is_english('Instagram'))
print(is_english('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(is_english('Docs To Go™ Free Office Suite'))
print(is_english('Instachat 😜'))


True
False
False
False


Note that we still want to keep in entries with emojis and characters such as tm.
We can edit our function to allow up to three characters with corresponding numbers falling outside the ASCII range. This way apps with emojis and special characters will still qualify.
The function is still not perfect but optimized to be fairly effective. 

We can do this by creating a variable 'non_acii' within the function. And during the loop, if any acii character goes above 127, we increment 'non_acii'. 
Then through an if statement we check to see if 'non_acii' is greater than or less than 3. 

In [14]:
def is_english(string):
    non_acii = 0
    
    for character in string:
        if ord(character) > 127:
            non_acii += 1
            
    if non_acii > 3:
        return False
    else:
        return True
    
    
print(is_english('Docs To Go™ Free Office Suite'))
print(is_english('Instachat 😜'))

True
True


## Now we can seperate our data into updated lists

With our optimized is_english function, we can create new lists that filter out non english apps.

In [15]:
android_english = []
ios_english = []

for app in android_clean:
    name = app[0]
    if is_english(name):
        android_english.append(app)
        
for app in ios:
    name = app[1]
    if is_english(name):
        ios_english.append(app)
        
explore_data(android_english,0,3,True)
print('\n')
explore_data(ios_english,0,3,True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows:  9614
Number of columns:  13


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+'

## Next we need to seperate the free apps

Since our data set contains both free and non free apps, we need to isolate the free apps for our analysis. 

### Final datasets for both Google and iOS

note that the price is in the 7th column of our Google data set.
We simply create a new list and run a for loop and append into our 'android_final' list if the price is equal to '0'

We do the same for our iOS data set.
Note that that the price in this data set is in the 4th column. 

In [16]:
ios_final = []
android_final = []

for app in android_english:
    price = app[7]
    if price == '0':
        android_final.append(app)
        
for app in ios_english:
    price = app[4]
    if price == '0.0':
        ios_final.append(app)
    
print(len(android_final))
print(len(ios_final))

8864
3222


## Data Analysis

Since we are ultimately looking for data to support our decision of what type of app would be successful on both Google play store and iOS app store, We can first look at what genres are most popular in both markets.

We can do this by building frequency tables for the 'prime_genre' column on the iOS App Store data set and the 'Genres' and 'Category'

We will design two functions:
    One to create the frequency tables that show percentages
    Another function to display the percentages in descending order

For the 'freq_table' we can create a dictionary named 'table' and an integer named 'total'.
From there we iterate through the dataset and increment total and add the values from the dataset into 'table'.
We can then create a second dictionary 'table_percentages' in which we can take the values from 'table' and multiply by 100 to get percentages. 

In [21]:
def freq_table(dataset, index):
    table = {}
    total = 0
    
    for row in dataset:
        total += 1
        value = row[index]
        if value in table:
            table[value] += 1
        else:
            table[value] = 1
            
    table_percentages = {}
    for key in table:
        percentage = (table[key] / total) * 100
        table_percentages[key] = percentage
        
    return table_percentages
    

In the 'display_table' function we run the previous 'freq_table' function and save it to a variable 'table'
Note that the sorted() prints an output on its own through built in functionality

In [19]:
def display_table(dataset, index):
    table = freq_table(dataset,index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)
        
    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':' , entry[0])

Note that the '-5' index means -5 from the end of the ios_header to land on 'prime_genre'

By displaying the iOS table:

We can see here that an overwhelming 58% of english apps that are also free are 'Games'. 
Entertainment follows second at about 8%

Hence we can get a clear idea that most users looking for free english apps are opting for 'Games'.
While more practical apps such as ducation, shopping, utilities, productivity, lifestyle, etc are very rare. 

In [22]:
print(ios_header)
display_table(ios_final, -5)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']
Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


Now if we take a look at which 'Categories' are most popular in the Google table:

We see that 'Family' sits at around 19%, followed by 'Game' at around 10% and 'Tools' at 8.5%.
The market seems significantly different on the Google Play Store.
Though if we investigate the 'Family' category on the Google Play Store we can notice that majority apps are actually games for children. 
Even Still there practical apps seem to be more popular on the Google Play Store. 

In [31]:
display_table(android_final, 1)

FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

One thing to note is that 'Genres' column is much more granular. Much of the data is split into more sub categories. Such as the genre of the game. 
For this reason we should opt to just use 'Category' since we are trying to get a more broad idea of the data. 


In [33]:
display_table(android_final, -4)

Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.591606498194946
Productivity : 3.892148014440433
Lifestyle : 3.892148014440433
Finance : 3.7003610108303246
Medical : 3.531137184115524
Sports : 3.463447653429603
Personalization : 3.3167870036101084
Communication : 3.2378158844765346
Action : 3.1024368231046933
Health & Fitness : 3.0798736462093865
Photography : 2.944494584837545
News & Magazines : 2.7978339350180503
Social : 2.6624548736462095
Travel & Local : 2.3240072202166067
Shopping : 2.2450361010830324
Books & Reference : 2.1435018050541514
Simulation : 2.0419675090252705
Dating : 1.861462093862816
Arcade : 1.8501805054151623
Video Players & Editors : 1.7712093862815883
Casual : 1.7599277978339352
Maps & Navigation : 1.3989169675090252
Food & Drink : 1.2409747292418771
Puzzle : 1.128158844765343
Racing : 0.9927797833935018
Role Playing : 0.9363718411552346
Libraries & Demo : 0.9363718411552346
Auto & Vehicles : 0.9250902527075

## Most Popular Apps by Genre on the iOS App Store

Since there is now 'Intalls' column to work with in the iOS data set, we can take the 'rating_count_tot' and create a frequency table with the 'Genres' to get a better idea of what the most popular apps are. 



In [23]:
print(ios_header)
print('\n')
genres_ios = freq_table(ios_final, -5)

for genre in genres_ios:
    total = 0
    len_genre = 0
    for app in ios_final:
        genre_app = app[-5]
        if genre_app == genre:
            # n_ratings is equal to rating_count_tot' column, this will the the total number of reviews for that genre
            n_ratings = float(app[5])
            # total will then become equal to n_rating
            total += n_ratings
            # each loop will then iterate len_genre, this will be the total number of apps in that genre
            len_genre += 1
    avg_n_ratings = total/ len_genre
    print(genre, ':', avg_n_ratings)
    
    

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


Social Networking : 71548.34905660378
Photo & Video : 28441.54375
Games : 22788.6696905016
Music : 57326.530303030304
Reference : 74942.11111111111
Health & Fitness : 23298.015384615384
Weather : 52279.892857142855
Utilities : 18684.456790123455
Travel : 28243.8
Shopping : 26919.690476190477
News : 21248.023255813954
Navigation : 86090.33333333333
Lifestyle : 16485.764705882353
Entertainment : 14029.830708661417
Food & Drink : 33333.92307692308
Sports : 23008.898550724636
Book : 39758.5
Finance : 31467.944444444445
Education : 7003.983050847458
Productivity : 21028.410714285714
Business : 7491.117647058823
Catalogs : 4004.0
Medical : 612.0


Though, one thing to be wary of is that our results here are skewed because of a few giants like Facebook, Google Maps, Waze, Skype, Pineterest etc.
These apps have a very high number of reviews. 

Our goal is to find popular genres, but these giant apps are making to seem that these genres are actually more popular than they are.

An example of this can be seen in the 'Refference' genre. Where two apps, 'Bible' and 'Dictionary.com' are skewing the average rating

We will sort this issue later on.

In [24]:
for app in ios_final:
    if app[-5] == 'Reference':
        print(app[1], ':', app[5])

Bible : 985920
Dictionary.com Dictionary & Thesaurus : 200047
Dictionary.com Dictionary & Thesaurus for iPad : 54175
Google Translate : 26786
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588
Merriam-Webster Dictionary : 16849
Night Sky : 12122
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693
GUNS MODS for Minecraft PC Edition - Mods Tools : 1497
Guides for Pokémon GO - Pokemon GO News and Cheats : 826
WWDC : 762
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 718
VPN Express : 14
Real Bike Traffic Rider Virtual Reality Glasses : 8
教えて!goo : 0
Jishokun-Japanese English Dictionary & Translator : 0


## Most Popular Apps by Genre on Google Play

In the Google Play data set we do have data on the number of installs, it just isn't very precise (100+, 1,000+, 5,000+, etc.)

In [25]:
display_table(android_final, 5)

1,000,000+ : 15.726534296028879
100,000+ : 11.552346570397113
10,000,000+ : 10.548285198555957
10,000+ : 10.198555956678701
1,000+ : 8.393501805054152
100+ : 6.915613718411552
5,000,000+ : 6.825361010830325
500,000+ : 5.561823104693141
50,000+ : 4.7721119133574
5,000+ : 4.512635379061372
10+ : 3.5424187725631766
500+ : 3.2490974729241873
50,000,000+ : 2.3014440433213
100,000,000+ : 2.1322202166064983
50+ : 1.917870036101083
5+ : 0.78971119133574
1+ : 0.5076714801444043
500,000,000+ : 0.2707581227436823
1,000,000,000+ : 0.22563176895306858
0+ : 0.04512635379061372
0 : 0.01128158844765343


Luckily we do not need precise data, we only need to get an idea of what apps genres attract the most users. 
We can leave the numbers as they are, which means that we'll consider that an app with 100,000+ installs has 100,000 installs, and an app with 1,000,000+ installs has 1,000,000 installs, and so on.

We use the same logic as we did with the iOS dataset. Except here we want to use the replace function to get rid of ',' , ''

In [26]:
categories_android = freq_table(android_final,1)

for category in categories_android:
    total = 0
    len_category = 0
    
    for app in android_final:
        category_app = app[1]
        if category_app == category:
            n_installs = app[5]
            n_installs = n_installs.replace(',', '')
            n_installs = n_installs.replace('+', '')
            total += float(n_installs)
            len_category += 1
            
    avg_n_installs = total / len_category
    print(category, ':', avg_n_installs)

ART_AND_DESIGN : 1986335.0877192982
AUTO_AND_VEHICLES : 647317.8170731707
BEAUTY : 513151.88679245283
BOOKS_AND_REFERENCE : 8767811.894736841
BUSINESS : 1712290.1474201474
COMICS : 817657.2727272727
COMMUNICATION : 38456119.167247385
DATING : 854028.8303030303
EDUCATION : 1833495.145631068
ENTERTAINMENT : 11640705.88235294
EVENTS : 253542.22222222222
FINANCE : 1387692.475609756
FOOD_AND_DRINK : 1924897.7363636363
HEALTH_AND_FITNESS : 4188821.9853479853
HOUSE_AND_HOME : 1331540.5616438356
LIBRARIES_AND_DEMO : 638503.734939759
LIFESTYLE : 1437816.2687861272
GAME : 15588015.603248259
FAMILY : 3695641.8198090694
MEDICAL : 120550.61980830671
SOCIAL : 23253652.127118643
SHOPPING : 7036877.311557789
PHOTOGRAPHY : 17840110.40229885
SPORTS : 3638640.1428571427
TRAVEL_AND_LOCAL : 13984077.710144928
TOOLS : 10801391.298666667
PERSONALIZATION : 5201482.6122448975
PRODUCTIVITY : 16787331.344927534
PARENTING : 542603.6206896552
WEATHER : 5074486.197183099
VIDEO_PLAYERS : 24727872.452830188
NEWS_AND_

From the data above we can see that the most installed category on the Google Play Store is 'Communication' with 38,456,119. This number is heavily skewed with a few apps with over one billion installs (WhatsApp, Facebook Messenger, Skype, Google Chrome, Gmail, and Hangouts).

We can get a better idea with the following for loop:

In [29]:
for app in android_final:
    if app[1] == 'COMMUNICATION' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
        print(app[0] , ':', app[5])


WhatsApp Messenger : 1,000,000,000+
imo beta free calls and text : 100,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Skype - free IM & video calls : 1,000,000,000+
Who : 100,000,000+
GO SMS Pro - Messenger, Free Themes, Emoji : 100,000,000+
LINE: Free Calls & Messages : 500,000,000+
Google Chrome: Fast & Secure : 1,000,000,000+
Firefox Browser fast & private : 100,000,000+
UC Browser - Fast Download Private & Secure : 500,000,000+
Gmail : 1,000,000,000+
Hangouts : 1,000,000,000+
Messenger Lite: Free Calls & Messages : 100,000,000+
Kik : 100,000,000+
KakaoTalk: Free Calls & Text : 100,000,000+
Opera Mini - fast web browser : 100,000,000+
Opera Browser: Fast and Secure : 100,000,000+
Telegram : 100,000,000+
Truecaller: Caller ID, SMS spam blocking & Dialer : 100,000,000+
UC Browser Mini -Tiny Fast Private & Secure : 100,000,000+
Viber Mess

If we filter out the apps with over 100 million installs, we are left with an average that is roughly 10 times less:

In [30]:
under_100_m = []

for app in android_final:
    n_installs = app[5]
    n_installs = n_installs.replace(',', '')
    n_installs = n_installs.replace('+', '')
    if (app[1] == 'COMMUNICATION') and (float(n_installs) < 100000000):
        under_100_m.append(float(n_installs))
        
sum(under_100_m) / len(under_100_m)

3603485.3884615386

We see the same pattern occur with 'Video Players' category, which is the runner up with 24,727,872 installs. The market is dominated by apps like Youtube, Google Play Movies & TV, or MX Player. 

This issue persists with many of the categories where a few giants skew the data and make it seem that the category is more popular than it is. Mainly due to the fact that these giants are very difficult to compete against. 

The 'Games' category seems popular, but the market seems very saturated. So competing here would be difficult as well.

We are trying to find a genre that will be profitable on both the play store and the app store. Lets take a look at 'Books and Refference' again since it has an average number of installs at 8,767,811.

In [32]:
for app in android_final:
    if app[1] == 'BOOKS_AND_REFERENCE':
        print(app[0], ':', app[5])

E-Book Read - Read Book for free : 50,000+
Download free book with green book : 100,000+
Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Free Panda Radio Music : 100,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
English Grammar Complete Handbook : 500,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
Google Play Books : 1,000,000,000+
AlReader -any text book reader : 5,000,000+
Offline English Dictionary : 100,000+
Offline: English to Tagalog Dictionary : 500,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
Recipes of Prophetic Medicine for free : 500,000+
ReadEra – free ebook reader : 1,000,000+
Anonymous caller detection : 10,000+
Ebook Reader : 5,000,000+
Litnet - E-books : 100,000+
Read books online : 5,000,000+
English to Urdu Dictionary : 500,000+
eBoox: book reader fb2 epub zip : 1,000,000+
English Persian Dictionary : 500,000+
Flybook : 500,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
E

The book and reference genre includes a variety of apps: software for processing and reading ebooks, various collections of libraries, dictionaries, tutorials on programming or languages, etc. It seems there's still a small number of extremely popular apps that skew the average:

In [33]:
for app in android_final:
      if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000,000+'
                                            or app[5] == '500,000,000+'
                                            or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

Google Play Books : 1,000,000,000+
Bible : 100,000,000+
Amazon Kindle : 100,000,000+
Wattpad 📖 Free Books : 100,000,000+
Audiobooks from Audible : 100,000,000+


Here the giants in the category seeem to be less dominant. We can further examine what the app installs look like somewhere in the middle for further analysis. 

In [34]:
for app in android_final:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000+'
                                            or app[5] == '5,000,000+'
                                            or app[5] == '10,000,000+'
                                            or app[5] == '50,000,000+'):
        print(app[0], ':', app[5])

Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
AlReader -any text book reader : 5,000,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
ReadEra – free ebook reader : 1,000,000+
Ebook Reader : 5,000,000+
Read books online : 5,000,000+
eBoox: book reader fb2 epub zip : 1,000,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
Moon+ Reader : 10,000,000+
English-Myanmar Dictionary : 1,000,000+
Golden Dictionary (EN-AR) : 1,000,000+
All Language Translator Free : 1,000,000+
Aldiko Book Reader : 10,000,000+
Dictionary - WordWeb : 5,000,000+
50000 Free eBooks & Free AudioBooks : 5,000,000+
Al-Quran (Free) : 10,000,000+
Al Quran Indonesia : 10,000,000+
Al'Quran Bahasa Indonesia : 10,000,000+
Al Quran Al karim : 1,000,000+
Al Quran : EAlim - Translations & MP3 Offline : 5,000,000+
Koran Read &MP3 30 Juz Offline : 1,000,000+
H

## Conclusion
'Books and References' seems to be dominated by software for processing and reading ebooks, as well as various collections of libraries and dictionaries, so building a similar app is not advisable. 

We can see that there are quite a few apps built around the Quran, which hints that building an app around a book could be profitable for both the Google Play Store and iOS App Store. 

The market does seem to be full of libraries, so adding a few features on top of simply the raw version might be benificial. 